src/graph/graphcommandexecutor.pas

```diff
@@ -44,37 +44,55 @@ procedure TGraphCommandExecutor.SaveGraph(ST: TCustomGraphCommand;
   SaveAction: TSaveGraphAction;
   Opt: TOption;
   Chart: TChart;
+  ChartPairs: TChartPairList;
+  Pair: TChartPair;
+  stratumValue: UTF8String;
+  aFileName: UTF8String;
+  allNames: UTF8String;
+  saveName: UTF8String;
+  saveMultiple: UTF8String;
 begin
   SaveAction := TSaveGraphAction.Create(nil);
-
+  ST.HasOption(['export', 'S', 'E'], Opt);
+  aFileName := Opt.Expr.AsString;
```

Comment:
I think this will fail hard if the option !E, !S or !export hasn't been used, since **Opt** will be a null pointer.

Afair - this should always be formatted as:
**if ST.HasOptions(...) then aFilename := Opt.Expr.AsString **

Reply:
I haven't tested with !e
Without the file name... I'll fix this.

Reply:
Ah... it's my bad. There is actually a check that such an option exists in the calling method. But it is good practice to check anyway, because that calling function might change with in the future and we do not remember to change this place too.

Reply:
all of these assignments are now only made if the option exists